src/EnergyPlus/SolarShading.cc

```diff
@@ -5540,7 +5540,7 @@ namespace SolarShading {
                     if (!Surface(BackSurfaceNumber).HeatTransSurf) continue;              // Skip non-heat transfer surfaces
                     if (Surface(BackSurfaceNumber).BaseSurf == GRSNR) continue;           // Skip subsurfaces of this GRSNR
                     if (BackSurfaceNumber == GRSNR) continue;                             // A back surface cannot be GRSNR itself
-                    if (Surface(BackSurfaceNumber).Zone != Surface(GRSNR).Zone) continue; // Skip if back surface not in zone
+                    if (Surface(BackSurfaceNumber).SolarEnclIndex != Surface(GRSNR).SolarEnclIndex) continue; // Skip if back surface not in same solar enclosure
```

Comment:
This is why beam solar wan't getting into the full enclosure, it was only going to the same zone as the window.

Reply:
Interesting, OK so you are matching against the enclosure index now, not the zone num.  :+1: